### PR TITLE
dirty_memory_manager: simplify, clarify, and document

### PR DESCRIPTION
**Pull request by avikivity**
This series undoes some recent damage to clarity, then
goes further by renaming terms around dirty_memory_manager
to be clearer. Documentation is added.

### CODE REVIEW

File: db/config.cc [hunk 1]
```diff
@@ -808,7 +808,7 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , prometheus_prefix(this, "prometheus_prefix", value_status::Used, "scylla", "Set the prefix of the exported Prometheus metrics. Changing this will break Scylla's dashboard compatibility, do not change unless you know what you are doing.")
     , abort_on_lsa_bad_alloc(this, "abort_on_lsa_bad_alloc", value_status::Used, false, "Abort when allocation in LSA region fails")
     , murmur3_partitioner_ignore_msb_bits(this, "murmur3_partitioner_ignore_msb_bits", value_status::Used, 12, "Number of most siginificant token bits to ignore in murmur3 partitioner; increase for very large clusters")
-    , virtual_dirty_soft_limit(this, "virtual_dirty_soft_limit", value_status::Used, 0.6, "Soft limit of virtual dirty memory expressed as a portion of the hard limit")
+    , unspooled_dirty_soft_limit(this, "unspooled_dirty_soft_limit", value_status::Used, 0.6, "Soft limit of unspooled dirty memory expressed as a portion of the hard limit")
```
Comment (denesb) [inline db/config.cc:811]: I think one of the reasons this was unused is that nobody understood what this option is. It doesn't even mention memtables.

**Review by denesb — APPROVED** [review]
The new term indeed achieves its goal -- it stop the reader from trying to understand how this all relates to virtual things -- spool starts with a clean slate here.
I think we need to remember to mention this config name change in the release notes. Not sure how we can make sure of that. Maybe mention it in the cover letter?

File: db/config.cc (hunk 1 above)
Comment (avikivity) [reply db/config.cc:811]: It's the type of options that we add when we want an escape hatch if something doesn't work. It did work, so the option wasn't used.

**Comment by annastuchlik** [comment]
@tzach Should this PR be backported to branch 5.1?

**Comment by tzach** [comment]
@annastuchlik yet another (nice) example for doc update as part of a code commit
No need to break it or backport just the doc part

**Pull request merged**